pkg/server/skydb/pq/device.go

```diff
@@ -24,16 +24,46 @@ import (
 )
 
 func (c *conn) GetDevice(id string, device *skydb.Device) error {
-	builder := psql.Select("type", "token", "user_id", "last_registered_at").
+	builder := psql.Select("type", "token", "user_id", "topic", "last_registered_at").
 		From(c.tableName("_device")).
-		Where("id = ?", id)
+		Where("id = ? AND topic IS NOT NULL", id)
```

Comment:
> The conclusion was something like: “need a migration to insert the device topic if they use token-based APNS”

If that is the case, I would make the following changes:

- the push handler will query all devices to be sent (by device ID or by user ID), no matter whether the `topic` is `NULL` or not.
- the push handler will filter the queried devices depending on whether the topic is specified on the request.
- the pusher will send push notification to each of the queried devices and the pusher will decide whether the notification can sent to that device.
    - the cert-based APNS pusher will respect the topic specified in the cert, instead of the one on device, so that it can send to the devices with `NULL` topic.
    - the token-based APNS pusher will use the topic on device and ignore those devices with `NULL` topic.

do you think it is okay ?